Makefile

```diff
@@ -16,47 +16,47 @@ PLATFORM_ID = $$( uname -s )
 endif
 PLATFORM = $$( \
 	case $(PLATFORM_ID) in \
-		( Linux | FreeBSD | OpenBSD | NetBSD | Windows) echo $(PLATFORM_ID) ;; \
+		( Linux | FreeBSD | OpenBSD | NetBSD | Windows | Darwin ) echo $(PLATFORM_ID) ;; \
 		( * ) echo Unrecognized ;; \
 	esac)
 
 CTAGS = $$( \
 	case $(PLATFORM_ID) in \
-		( Linux ) echo "ctags" ;; \
+		( Linux | Darwin ) echo "ctags" ;; \
 		( FreeBSD | OpenBSD | NetBSD | Windows) echo "exctags" ;; \
 		( * ) echo Unrecognized ;; \
 	esac)
 
 MAKE = $$( \
 	case $(PLATFORM_ID) in \
-		( Linux | Windows) echo "make" ;; \
+		( Linux | Windows | Darwin ) echo "make" ;; \
 		( FreeBSD | OpenBSD | NetBSD ) echo "gmake" ;; \
 		( * ) echo Unrecognized ;; \
 	esac)
 
 MESON = $$( \
 	case $(PLATFORM_ID) in \
-		( Linux | Windows) echo "meson" ;; \
+		( Linux | Windows | Darwin ) echo "meson" ;; \
 		( FreeBSD | OpenBSD | NetBSD ) echo "meson" ;; \
 		( * ) echo Unrecognized ;; \
 	esac)
 
 NPROC = $$( \
 	case $(PLATFORM_ID) in \
-		( Linux | Windows) nproc ;; \
-		( FreeBSD | OpenBSD | NetBSD ) sysctl -n hw.ncpu ;; \
+		( Linux | Windows ) nproc ;; \
+		( FreeBSD | OpenBSD | NetBSD | Darwin ) sysctl -n hw.ncpu ;; \
 		( * ) echo Unrecognized ;; \
 	esac)
 
 GIT = $$( \
 	case $(PLATFORM_ID) in \
-		( Linux | Windows) echo "git" ;; \
+		( Linux | Windows | Darwin ) echo "git" ;; \
 		( FreeBSD | OpenBSD | NetBSD ) echo "git" ;; \
 		( * ) echo Unrecognized ;; \
 	esac)
 
 PROJECT_VER = $(( python3 scripts/xnvme_ver.py --path meson.build ))
-BUILD_DIR?=builddir
+BUILD_DIR?=builddir-$(shell uname | tr '[:upper:]' '[:lower:]')-$(shell arch)
```

Comment:
A brief description in the commit-message describing this would be great.
This would create a builddir named ``builddir-linux-x86_64`` on Linux and something fitting on MacOS.
Why is that required?

Reply:
Or revert simply use the "builddir" name...

Reply:
I'll withdraw it for now, and maybe do a separate PR with it, because the functionality is not complete (subprojects are not acting nice). But it allows you to compile for multiple platforms/archs in the same repo.

Reply:
I see that would be useful, but yeah, let's do that in another set of changes which has more of that functional completeness :)